cloudinit/config/cc_write_files.py

```diff
@@ -163,13 +190,15 @@
 
 
 def handle(name, cfg, _cloud, log, _args):
-    files = cfg.get('write_files')
-    if not files:
+    validate_cloudconfig_schema(cfg, schema)
+    file_list = cfg.get('write_files', [])
+    is_not_deferred_file = lambda f: f.get('defer', DEFAULT_DEFER) == False
+    filtered_files = list(filter(is_not_deferred_file, file_list))
```

Comment:
I think something like this might be a little clearer:
```python
    non_deferred_files = [
        f for f in file_list if not util.get_cfg_option_bool(f, 'defer')]
```
(non_deferred_files replacing filtered_files)

Reply:
I like the functional approach better, but sure. Though, I would keep `f.get('defer', DEFAULT_DEFER)`. I believe there is no need for `util.get_cfg_option_bool`, since the schema validation happens before and would already yell if the type of `defer` is not correct.

Or is there some weird behavior about `.get` that a Python armature such as myself is not aware of?

Reply:
> I believe there is no need for util.get_cfg_option_bool, since the schema validation happens before and would already yell if the type of defer is not correct.
Or is there some weird behavior about .get that a Python armature such as myself is not aware of?

"Yell" in this context means log a warning and then keep executing with the wrong data. If somebody has a cloud-config with something like:
```yaml
write_files:
  ...
  defer: 'false'
```
it's better to  log a warning and do what the user intended rather than log a warning and do the wrong thing.

Reply:
Oh, I'm sorry. Without actually checking, I was somehow assuming that the behavior would be as if `def validate_cloudconfig_schema( ..., strict=True)`.

I understand your reasoning. I don't fully agree though, but that doesn't matter here :). I pushed another change that make use of `util.get_cfg_option_bool`.